### ISSUE TITLE
The issue of discrepancy between test results and paper results for BCH(31,16)

### ISSUE THREAD

**Issue by helinru** [yoniLc/E2E_DC_ECCT#2]
Hello author:
I trained a (31,16) E2E_DC_ECCT using your code with the following settings: learning rate 1e−4, cosine decay scheduler down to 1e−6, |Ω| ≤ 1, Ω stopped at iteration 800, N = 2, d = 32, and Ω0 was randomly sampled.
However, the test results I obtained are Test -ln(BER) 4: 5.086e+00, 5: 6.487e+00, 6: 8.299e+00.
In your paper's Table 1, the BCH (31,16) results are 5.30, 6.89, 9.09, which is a significant difference. Yet they are somewhat close to the results in Table 5 (5.16, 6.51, 8.19). I'm confused as to why this discrepancy has occurred.
thanks for the reply！
helinru

**Comment by GGuuu** [comment]
I have same problem!

**Comment by yoniLc** [comment]
Hi, it's hard to tell as the code provided does not include reproduction scripts so some discrepancy can exist with your implementation.

The settings of the main results and the ablation study differ (e.g., regular STE was used for ablation). In order to reproduce the main results, please follow 4.5.

The initialization of \Omega (and the STE hyperparameters) can also have a substantial impact, so I would first try using different seeds.

**Comment by helinru** [comment]
Thank you for your suggestions.
